crates/node/service/src/actors/derivation.rs

```diff
@@ -265,11 +262,10 @@ where
             return Ok(());
         }
 
-        // The L2 Safe Head must be advanced before producing new payload attributes.
-        if self.engine_l2_safe_head.borrow().block_info.number <=
-            self.l2_safe_head.block_info.number
-        {
-            debug!(target: "derivation", engine_safe_head = ?self.engine_l2_safe_head.borrow().block_info.number, l2_safe_head = ?self.l2_safe_head.block_info.number, "L2 safe head unchanged");
+        // Wait for the engine to initialize unknowns prior to kicking off derivation.
+        let engine_safe_head = *self.engine_l2_safe_head.borrow();
+        if engine_safe_head.block_info.hash.is_zero() {
```

Comment:
We also need to add a condition here that the safe head has actually changed. Can use the "has seen" functionality of the watch channel for this thankfully :D A combo of [`has_changed`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.has_changed) + [`borrow_and_update`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html#method.borrow_and_update) will do the trick.